src/pages/TrueSightV2/pages/TokenAnalysisList.tsx

```diff
@@ -0,0 +1,1116 @@
+        setMenuLeft(Math.min(left, rowWidth - menuWidth))
+        setShowMenu(true)
+      }
+    } else {
+      navigate(`${APP_PATHS.KYBERAI_EXPLORE}/${token.tokens[0].chain}/${token.tokens[0].address}`, {
+        state: { from: location },
+      })
+    }
+  }
+
+  const handleWatchlistClick = (e: any) => {
+    e.stopPropagation()
+    if (!account) return
+    setLoadingStar(true)
+    if (isWatched) {
+      mixpanelHandler(MIXPANEL_TYPE.KYBERAI_ADD_TOKEN_TO_WATCHLIST, {
+        token_name: token.symbol?.toUpperCase(),
+        source: KYBERAI_LISTYPE_TO_MIXPANEL[listType],
+        ranking_order: index,
+        option: 'remove',
+      })
+      Promise.all(
+        token.tokens.map(t => removeFromWatchlist({ wallet: account, tokenAddress: t.address, chain: t.chain })),
+      ).then(() => {
+        setIsWatched(false)
+        setLoadingStar(false)
+      })
+    } else {
+      mixpanelHandler(MIXPANEL_TYPE.KYBERAI_ADD_TOKEN_TO_WATCHLIST, {
+        token_name: token.symbol?.toUpperCase(),
+        source: KYBERAI_LISTYPE_TO_MIXPANEL[listType],
+        ranking_order: index,
+        option: 'add',
+      })
+      Promise.all(
+        token.tokens.map(t => addToWatchlist({ wallet: account, tokenAddress: t.address, chain: t.chain })),
+      ).then(() => {
+        setIsWatched(true)
+        setLoadingStar(false)
+      })
+    }
+  }
+
+  useEffect(() => {
+    setIsWatched(token.isWatched)
+  }, [token.isWatched])
+
+  const latestKyberScore: IKyberScoreChart | undefined = token?.ks_3d?.[token.ks_3d.length - 1]
+  return (
+    <tr key={token.sourceTokenId} ref={rowRef} onClick={handleRowClick} style={{ position: 'relative' }}>
+      <td>
+        <RowFit gap="6px">
+          {
+            <SimpleTooltip text={isWatched ? t`Remove from watchlist` : t`Add to watchlist`} hideOnMobile>
+              <StarWithAnimation
+                watched={isWatched}
+                loading={loadingStar}
+                onClick={handleWatchlistClick}
+                size={above768 ? 20 : 16}
+              />
+            </SimpleTooltip>
+          }
+          {above768 ? index : <></>}
+        </RowFit>
+      </td>
+      <td className={isScrolling ? 'table-cell-shadow-right' : ''}>
+        <Row gap="8px">
+          <div style={{ position: 'relative', width: '36px', height: '36px' }}>
+            <img
+              alt="tokenInList"
+              src={token.tokens[0].logo}
+              width="36px"
+              height="36px"
+              loading="lazy"
+              style={{ borderRadius: '18px' }}
+            />
+          </div>
+
+          <Column gap="8px" style={{ cursor: 'pointer', alignItems: 'flex-start' }}>
+            <Text style={{ textTransform: 'uppercase' }}>{token.symbol}</Text>{' '}
+            <RowFit gap="6px" color={theme.text}>
+              {token.tokens.map(item => {
+                if (item.chain === 'ethereum') return <Icon key="eth-mono" id="eth-mono" size={12} title="Ethereum" />

```

Comment:
duplicate
<img width="597" alt="image" src="https://github.com/KyberNetwork/kyberswap-interface/assets/33005392/f1c85caa-b081-44eb-aff2-8f2d19b93267">

Reply:
yes, but there is some difference inside those codes. Hard to reuse it, it's ok to leave it right now.